### PR TITLE
Update pandas.append to pandas.concat due to deprecation

### LINKED ISSUE #1971: Deprecated Frame.append and Series.append in Pandas
**Describe the bug**
FutureWarning about `.append()` deprecation. From pandas [documentation](https://pandas.pydata.org/docs/whatsnew/v1.4.0.html):
> DataFrame.append() and Series.append() have been deprecated and will be removed in Pandas 2.0. Use pandas.concat() instead (GH35407).

**To Reproduce**
```python
az.compare({
    'm1': trace_1,
    'm2: trace_2,
})
```

```python
~/miniconda3/lib/python3.9/site-packages/arviz/stats/stats.py:212: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  ics = ics.append([ic_func(dataset, pointwise=True, scale=scale)])
~/miniconda3/lib/python3.9/site-packages/arviz/stats/stats.py:212: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  ics = ics.append([ic_func(dataset, pointwise=True, scale=scale)])
```

**Additional context**
```
ArviZ version: 0.11.2
Pandas version: 1.4.0
```

### PR DESCRIPTION
**Pull request by aliakbars**
## Description

Fix FutureWarning about `DataFrame.append()` deprecation. (closes #1971)

## Checklist

- [x] Follows [official](https://github.com/arviz-devs/arviz/blob/main/CONTRIBUTING.md#pull-request-checklist) PR format
- [x] Code style  correct (follows pylint and black guidelines)
- [x] Changes are listed in [changelog](https://github.com/arviz-devs/arviz/blob/main/CHANGELOG.md#v0xx-unreleased)

### CODE REVIEW

**Comment by OriolAbril** [comment]
The behaviour seems to have changed, have you checked locally what happens? Could you maybe share the outputs from the docs examples https://arviz-devs.github.io/arviz/api/generated/arviz.compare.html so we can see what has changed?

**Comment by aliakbars** [comment]
```python
import arviz as az
data1 = az.load_arviz_data("non_centered_eight")
data2 = az.load_arviz_data("centered_eight")
compare_dict = {"non centered": data1, "centered": data2}
az.compare(compare_dict)
```

```python
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Input In [6], in <module>
      4 data2 = az.load_arviz_data("centered_eight")
      5 compare_dict = {"non centered": data1, "centered": data2}
----> 6 az.compare(compare_dict)

File /lustre/home/ucakase/workspace/arviz/arviz/stats/stats.py:253, in compare(dataset_dict, ic, method, b_samples, alpha, seed, scale, var_name)
    251         raise e.__class__(f"Encountered error trying to compute {ic} from model {name}.") from e
    252 print(type(ics), ics.shape)
--> 253 ics.index = names
    254 ics.sort_values(by=ic, inplace=True, ascending=ascending)
    255 ics[ic_i] = ics[ic_i].apply(lambda x: x.values.flatten())

File ~/miniconda3/lib/python3.9/site-packages/pandas/core/generic.py:5596, in NDFrame.__setattr__(self, name, value)
   5594 try:
   5595     object.__getattribute__(self, name)
-> 5596     return object.__setattr__(self, name, value)
   5597 except AttributeError:
   5598     pass

File ~/miniconda3/lib/python3.9/site-packages/pandas/_libs/properties.pyx:70, in pandas._libs.properties.AxisProperty.__set__()

File ~/miniconda3/lib/python3.9/site-packages/pandas/core/generic.py:769, in NDFrame._set_axis(self, axis, labels)
    767 def _set_axis(self, axis: int, labels: Index) -> None:
    768     labels = ensure_index(labels)
--> 769     self._mgr.set_axis(axis, labels)
    770     self._clear_item_cache()

File ~/miniconda3/lib/python3.9/site-packages/pandas/core/internals/managers.py:214, in BaseBlockManager.set_axis(self, axis, new_labels)
    212 def set_axis(self, axis: int, new_labels: Index) -> None:
    213     # Caller is responsible for ensuring we have an Index object.
--> 214     self._validate_set_axis(axis, new_labels)
    215     self.axes[axis] = new_labels

File ~/miniconda3/lib/python3.9/site-packages/pandas/core/internals/base.py:64, in DataManager._validate_set_axis(self, axis, new_labels)
     61     pass
     63 elif new_len != old_len:
---> 64     raise ValueError(
     65         f"Length mismatch: Expected axis has {old_len} elements, new "
     66         f"values have {new_len} elements"
     67     )

ValueError: Length mismatch: Expected axis has 18 elements, new values have 2 elements
```

File: arviz/stats/stats.py [hunk 1]
```diff
@@ -246,7 +246,7 @@ def compare(
         try:
             # Here is where the IC function is actually computed -- the rest of this
             # function is argument processing and return value formatting
-            ics = ics.append([ic_func(dataset, pointwise=True, scale=scale, var_name=var_name)])
+            ics = pd.concat([ics, ic_func(dataset, pointwise=True, scale=scale, var_name=var_name)])
```
Comment (MarcoGorelli) [inline arviz/stats/stats.py:249]: I think you can replace this whole piece with something like
```
ics = pd.concat({key: ic_func(val, pointwise=True, scale=scale, var_name=var_name) for key, val in dataset_dict.items()}, axis=1).T
```
as part of the reason for deprecating `append` is that it's slow

Comment (aliakbars) [reply arviz/stats/stats.py:249]: What about the exception?

Comment (MarcoGorelli) [reply arviz/stats/stats.py:249]: you could append these internal results to some other `dict` and then call `concat` at the end

or make a wrapped around `ic_func` which raises the exception if it fails

File: arviz/stats/stats.py [hunk 2]
```diff
@@ -239,16 +239,18 @@ def compare(
...
         try:
             # Here is where the IC function is actually computed -- the rest of this
             # function is argument processing and return value formatting
-            ics = ics.append([ic_func(dataset, pointwise=True, scale=scale, var_name=var_name)])
+            ics[name] = ic_func(dataset, pointwise=True, scale=scale, var_name=var_name)
         except Exception as e:
             raise e.__class__(f"Encountered error trying to compute {ic} from model {name}.") from e
+
+    ics = pd.concat(ics, axis=1).T
     ics.index = names
```
Comment (MarcoGorelli) [inline arviz/stats/stats.py:254]: I think this line might be unnecessary now?

Comment (aliakbars) [reply arviz/stats/stats.py:254]: I am replacing the initialisation of `ics` from `pd.DataFrame()` to `dict`. So this line makes it a DataFrame.

Comment (MarcoGorelli) [reply arviz/stats/stats.py:254]: sorry I meant the line after (`ics.index = names`)

Comment (aliakbars) [reply arviz/stats/stats.py:254]: Right, yes. I'll delete it.

File: arviz/stats/stats.py [hunk 3]
```diff
@@ -239,17 +239,16 @@ def compare(
...
         try:
             # Here is where the IC function is actually computed -- the rest of this
             # function is argument processing and return value formatting
-            ics = ics.append([ic_func(dataset, pointwise=True, scale=scale, var_name=var_name)])
+            ics[name] = ic_func(dataset, pointwise=True, scale=scale, var_name=var_name)
         except Exception as e:
             raise e.__class__(f"Encountered error trying to compute {ic} from model {name}.") from e
-    ics.index = names
+
+    ics = pd.concat(ics, axis=1).T
```
Comment (ahartikainen) [inline arviz/stats/stats.py:251]:
```suggestion
    ics =  pd.DataFrame.from_dict(ics, orient="index")
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -12,6 +12,7 @@
 * Remove manual setting of 2d KDE limits ([1939](https://github.com/arviz-devs/arviz/pull/1939))
 * Pin to bokeh<3 version ([1954](https://github.com/arviz-devs/arviz/pull/1954))
 * Fix legend labels in plot_ppc to reflect prior or posterior. ([1967](https://github.com/arviz-devs/arviz/pull/1967))
+* Change `DataFrame.append` to `pandas.concat` ([1973](https://github.com/arviz-devs/arviz/pull/1973))
```
Comment (MarcoGorelli) [inline CHANGELOG.md:15]: I don't know what's typical in ArviZ, but are non-user-facing changes typically documented?

Comment (ahartikainen) [reply CHANGELOG.md:15]: It is fine

**Pull request merged**